app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
+
+#  News Articles
+#
+#  Actions
+#
+#    create:           Create article from data in "new" form
+#    destroy::         Destroy article
+#    edit::            Display form for editing article
+#    index::           List all articles
+#    index_articles::  List selected (based on last search) articles
+#    new::             Display form for new article
+#    show::            Show one article
+#    update::          Update article from "edit" form
+#
+#
+#  Callbacks and Methods
+#
+#    ignore_request_unless_permitted:: Unless user permitted to perform request,
+#                       then index_articles
+#    permitted?         boolean: permitted to create/update/destroy Articles
+#
+class ArticlesController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :index_articles,
+    :list_articles, # aliased
+    :show,
+    :show_article # aliased
+  ]
+  before_action :store_location
+  before_action :ignore_request_unless_permitted, except: [
+    :index,
+    :index_articles,
+    :list_articles, # aliased
+    :show,
+    :show_article # aliased
+  ]
+
+  ############ - Actions to Display data (index, show, etc.)
```

Comment:
I suggest that all controllers uniformly order actions and methods. (Makes it easier to find things and read the code.)
There isn't an official Rails default ordering. (Generated controllers are empty. See [here](https://guides.rubyonrails.org/getting_started.html#laying-down-the-groundwork).) So I used the same order (indicted by the comments in the file) as AWDR:
1. Actions to Display Data (index, show, etc.)
2. Actions to Present Forms -- (new, edit, etc.)
3. Actions to Modify data: (create, update, destroy, etc.)
4. Public methods (unrouted)
5. private methods
This may also suggest a way to divide the controller into separate files, if that's desirable.